.github/workflows/publish-crate.yml

```diff
@@ -0,0 +1,45 @@
+## This Source Code Form is subject to the terms of the Mozilla Public
+## License, v. 2.0. If a copy of the MPL was not distributed with this
+## file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+name: Publish Crate Workflow
+
+on:
+  push:
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+'
+
+    # ?? Should this run on merge/push to master instead of main (guessing you are using main like develop)?
+    # branches:
+    #   - main
```

Comment:
Note: `master` doesn't exist in this repo (`main` is just the new default branch for more recent git versions).

thought: I'm really not sure we want to publish automatically to main.  As well as whether we want to do this, if the version of the package we are publishing has not changed in the new commit, the `cargo publish` command will fail, as the version already exists in the package repository (crates.io).

Reply:
There is an action that handles some additional checks, but was hesitant to use it as it's not that popular. Might be worth a look: https://github.com/katyo/publish-crates

In that case can leave this to be a dispatch trigger only for now (manually).